### DISCUSSION TITLE
Does "forge test" run the tests against an instance of REVM?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [foundry-rs/foundry#3898] [General]
What is the EVM against which tests are run when users run the `forge test` command?

`forge test --help` does not specify this:

> Run the project's tests.

Nor I could find any information about this in the README.

**Comment by samtay** [comment]
This is my understanding. You can see that forge [imports its EVM backend here](https://github.com/foundry-rs/foundry/blob/3861b3f7d6e173cf709e4a99c1584db92fbd6c63/forge/src/lib.rs#L23-L24), and if you look at the [forge/Cargo.toml](https://github.com/foundry-rs/foundry/blob/3861b3f7d6e173cf709e4a99c1584db92fbd6c63/forge/Cargo.toml#L11) you'll see this resolves to the [evm crate in this workspace](https://github.com/foundry-rs/foundry/tree/3861b3f7d6e173cf709e4a99c1584db92fbd6c63/evm). Finally, if you look there, it seems like its mostly just a wrapper around `revm`.

**Reply by PaulRBerg** [reply]
Thanks for your comment, @samtay.

It does indeed seem that Forge is running the REVM on `forge test`.
